pulsar/schema.go

```diff
@@ -125,20 +126,19 @@ func NewSchema(schemaType SchemaType, schemaData []byte, properties map[string]s
 }
 
 type AvroCodec struct {
-	Codec *goavro.Codec
+	Codec avro.Schema
 }
 
-func NewSchemaDefinition(schema *goavro.Codec) *AvroCodec {
-	schemaDef := &AvroCodec{
+func NewSchemaDefinition(schema avro.Schema) *AvroCodec {
```

Comment:
I worry that this method will be called on the outside(users).

Reply:
Should this function not be exported ?

Reply:
This parameter has been changed to `avro.Schema`.